yt/frontends/gadget/data_structures.py

```diff
@@ -350,7 +356,7 @@ def _set_code_unit_attributes(self):
         self.specific_energy_unit = self.quan(*specific_energy_unit)
 
     @staticmethod
-    def _validate_header(filename):
+    def _validate_header(filename, header_size=256):
```

Comment:
I have a question here. Currently I'm setting default header size to be 256, which would then be used in `GadgetDataset._is_valid`. This means datasets with non-standard headers would not be able to be loaded using `yt.load`. Instead, they have to be loaded using `GadgetDataset` directly.

I'm thinking about another method to solve this. In the endianness check below, we could do something like:
```python
def endian_check(f):
    for endianswap in '<>':
        f.seek(0)
        header_size = struct.unpack(endianswap + 'I', f.read(4))
        f.seek(4 + header_size)
        header_size_check = struct.unpack(endianswap + 'I', f.read(4))
        if header_size == header_size_check:
            return endianswap
    return False
```
This is only a consistency check, which shall work for any header size. But the drawback is that there could be false positives (although these should be rare cases).

Any suggestions?

Reply:
One option would be to detect this case and emit a loud warning if it happens saying the user is doing something nonstandard and yt had to guess a bit